src/test/java/SearchServletTest.java

```diff
@@ -0,0 +1,164 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.SearchServlet;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class SearchServletTest {
+  private static final String EMPTY_STRING = "";
+  private static final String NULL_VALUE = null;
+
+  private static final String NULL_EXCEPTION_MESSAGE =
+      "Null Field: Receipt unable to be queried at this time, please try again.";
+  private static final String NUMBER_EXCEPTION_MESSAGE =
+      "Invalid Price: Receipt unable to be queried at this time, please try again.";
+  private static final String PARSE_EXCEPTION_MESSAGE =
+      "Dates Unparseable: Receipt unable to be queried at this time, please try again.";
+
+  // Values for a valid test.
+  private static final String CST_TIMEZONE_ID = "America/Chicago";
+  private static final String CATEGORIES = "drink";
+  private static final String SHORT_DATE_RANGE = "February 1, 2003 - February 28, 2003";
+  private static final String LONG_DATE_RANGE = "January 1, 2010 - July 31, 2020";
+  private static final String STORE = "walmart";
+  private static final String MIN_PRICE = "5.00";
+  private static final String MAX_PRICE = "30.00";
+
+  // Local Datastore
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true);
+
+  @Mock private SearchServlet servlet;
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  private DatastoreService datastore;
+  private StringWriter stringWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    servlet = new SearchServlet(datastore);
+
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void queryWithAllFieldsFilled() throws IOException {
+    // Add mock receipts to datastore.
```

Comment:
Can we add a comment section explaining the setup of this test. Similar to the comments in the TDD unit tests. It's difficult to understand the test because there are multiple receipts and search parameters. For example, could you explain why the walmart receipt was returned.

It would be great if this could be done for all these unit tests. Let me know if you have any questions.